bql/grammar/grammar_test.go

```diff
@@ -271,6 +301,31 @@ func TestAcceptOpsByParseAndSemantic(t *testing.T) {
 		{`insert data into ?a {/_<foo> "bar"@[] /_<foo> .
 				                      /_<foo> "bar"@[] "bar"@[1975-01-01T00:01:01.999999999Z] .
 				                      /_<foo> "bar"@[] "yeah"^^type:text};`, 1, 3},
+	}
+	p, err := NewParser(SemanticBQL())
+	if err != nil {
+		t.Errorf("grammar.NewParser: Should have produced a valid BQL parser, %v", err)
+	}
+	for _, entry := range table {
+		st := &semantic.Statement{}
+		if err := p.Parse(NewLLk(entry.query, 1), st); err != nil {
+			t.Errorf("Parser.consume: Failed to accept entry %q with error %v", entry, err)
+		}
+		if got, want := len(st.OutputGraphNames()), entry.graphs; got != want {
+			t.Errorf("Parser.consume: Failed to collect right number of graphs for case %v; got %d, want %d", entry, got, want)
+		}
+		if got, want := len(st.Data()), entry.triples; got != want {
+			t.Errorf("Parser.consume: Failed to collect right number of triples for case %v; got %d, want %d", entry, got, want)
+		}
+	}
+}
+
+func TestAcceptDeleteDataFromGraphOpsByParseAndSemantic(t *testing.T) {
+	table := []struct {
+		query   string
+		graphs  int
```

Comment:
I'd rather suggest to use []string{"?graph",...} instead of testing for length. Here and on the other tests. Since this aso improves on correctness.